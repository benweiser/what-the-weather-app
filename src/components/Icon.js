import React from "react";
import "../styles/icons/weather-icons.min.css";
import styled, { css } from "react-emotion";

const iconSizes = {
  sm: "24px",
  md: "32px",
  lg: "48px",
  xl: "64px"
};

const StyledIcon = styled("span")`
  font-size: ${props => iconSizes[props.size] || props.size};
`;

const Icon = props => {
  return <StyledIcon className={`wi ${props.name}`} {...props} />;
};

export default Icon;
