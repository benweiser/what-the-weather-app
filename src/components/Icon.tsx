import React from 'react';
import 'styled-components/macro';
import '../styles/icons/weather-icons.min.css';

export interface IconProps {
  name: string;
  size: string;
}

const iconSizes: { [key: string]: string } = {
  sm: '24px',
  md: '32px',
  lg: '48px',
  xl: '64px'
};

const Icon = (props: IconProps) => {
  const { name, size, ...remainingProps } = props;
  return (
    <span
      className={`wi ${name}`}
      css={`
        font-size: ${iconSizes[size] || props.size};
      `}
      {...remainingProps}
    />
  );
};

export default Icon;
