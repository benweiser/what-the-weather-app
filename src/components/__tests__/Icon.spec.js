import renderer from 'react-test-renderer';
import * as emotion from 'emotion';
import { createSerializer } from 'jest-emotion';
import React from 'react';
import Icon from '../Icon';

expect.addSnapshotSerializer(createSerializer(emotion));

describe('Icon component', () => {
  test('renders', () => {
    const component = renderer.create(<Icon name="wi test" size="xl" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
