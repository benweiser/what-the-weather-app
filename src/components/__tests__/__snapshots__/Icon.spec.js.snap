// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component renders 1`] = `
.emotion-0 {
  font-size: 64px;
}

<span
  className="wi wi test emotion-0"
  size="xl"
/>
`;
